front/src/app/todo-list/todo-list.component.ts

```diff
@@ -48,21 +49,61 @@ export class TodoListComponent implements OnInit, OnDestroy {
 
   taskListInputValue: string = '';
 
+  resize$ = fromEvent(window, 'resize')
+
+  sidebarMode: MatDrawerMode = 'side'
+
+  sidebarOpen: string = 'true'
+
   private destroy$ = new Subject<void>();
 
   constructor(private taskListManager: TaskListManagerService, private apiService: ApiService, private changeDetectorRef: ChangeDetectorRef, private router: Router) {}
 
   ngOnInit(): void {
+    this.sidebarOpen = (localStorage.getItem('sidebarOpen') as string)
+    if (document.documentElement.clientWidth <= 750) {
+      this.sidebarMode = 'over'
+    }
+
+    this.resize$
+      .subscribe(() => {

```

Comment:
1. no take until
2. you can get rid of extra check above (which sets initial sidebarMode value) by using rxjs `startWith` operator